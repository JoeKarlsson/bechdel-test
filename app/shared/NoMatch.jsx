/* jshint esversion: 6 */
'use strict';

import React from 'react';

export default React.createClass({
  render() {
    return <div>No Page Found</div>
  }
});