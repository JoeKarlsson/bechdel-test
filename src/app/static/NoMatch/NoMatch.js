import React from 'react';

const NoMatch = () => (
	<div>No Page Found</div>
);

export default NoMatch;
