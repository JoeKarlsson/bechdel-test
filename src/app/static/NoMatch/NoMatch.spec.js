import React from 'react';
import {
	shallow,
	configure,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import NoMatch from './NoMatch';

configure({ adapter: new Adapter() });

describe('NoMatch', () => {
	let wrapper;

	global.requestAnimationFrame = (callback) => {
		setTimeout(callback, 0);
	};

	beforeEach(() => {
		wrapper = shallow(<NoMatch />);
	});

	describe('rendering', () => {
		describe('initial state', () => {
			it('is rendered', () => {
				const component = renderer.create(<NoMatch />);
				const tree = component.toJSON();
				expect(tree).toMatchSnapshot();
			});

			it('is rendered correctly', () => {
				expect(wrapper).toHaveLength(1);
			});

			it('should have correct inital instance', () => {
				const initialInstance = wrapper.instance();
				const expectedInstance = null;
				expect(initialInstance).toBe(expectedInstance);
			});
		});
	});
});
