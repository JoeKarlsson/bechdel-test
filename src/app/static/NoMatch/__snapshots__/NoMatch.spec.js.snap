// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch rendering initial state is rendered 1`] = `
<div>
  No Page Found
</div>
`;
